### DISCUSSION TITLE
the  successful tests took 0 milliseconds to run?

### DISCUSSION THREAD

**Discussion by Huixvan-Li** [google/googletest#4498] [Community Help]
Why does the console output and generated xml report after a successful test run indicate that the test took 0 milliseconds to run? Is this a bug？

**Comment by Shriharsh-Deshmukh** [comment]
No, it's not a bug. The console output and generated XML report showing `0 milliseconds` typically occurs when the test execution time is too small to be captured by the system clock's precision. In most cases, test durations are rounded down to `0 ms` if they are extremely short, often due to fast tests or limited clock resolution.

If more precise timing is critical, you may need to use high-resolution timers or run the test multiple times in a loop to accumulate measurable time.
